src/main.js

```diff
@@ -337,58 +134,15 @@ Apify.main(async () => {
         },
         proxyConfiguration: proxyConfig,
         preNavigationHooks: [async ({ request, page }, gotoOptions) => {
-            const userAgent = headerGenerator.getHeaders()['user-agent'];
+            /** @type {any} */
+            const headers = headerGenerator.getHeaders();
```

Comment:
This is great blocking mitigation thinking! There is no need to change it if it works, I want to provide a bit of context for this. Changing the user-agent for a browser, it a bit more complex. This way, you would change the user-agent in the browser request headers. However, different browser versions might have different browser features. Some advance protections could detect this lie and block it. If you need to overcome this kind of browser detection system from SDK v2.2, a fingerprint generator is included. The fingerprint logic, among other attributes, uses a different userAgent per session. To turn it on, you need to pass `{useFingerprints: true} `to `browserPoolOptions` Puppeteer crawler options

Reply:
I replaced manual user-agent settings with `useFingerprints` as you suggested and it seems to be working well, no blocking or crashes. Thanks for the tip! 

Now I can also remove the whole `HeaderGenerator` with mysterious `operatingSystems: process.platform === 'win32'` settings you mentioned earlier.